src/zxing/zxing/MultiFormatReader.cpp

```diff
@@ -96,45 +97,57 @@ void MultiFormatReader::enableReaders(zxing::DecodeHints hints, bool allowAll)
       hints.containsFormat(BarcodeFormat::RSS_14) ||
       hints.containsFormat(BarcodeFormat::RSS_EXPANDED);
 
-  #ifdef ENABLE_DECODER_1D_BARCODES
     if ((allowAll || addOneDReader) && !tryHarder) {
+  #ifdef ENABLE_DECODER_1D_BARCODES
       readers_.push_back(Ref<Reader>(new zxing::oned::MultiFormatOneDReader(hints)));
-    }
+  #else
+      qWarning("Failed to enable 1D Barcodes because they are disabled on compile time.");
```

Comment:
All the qWarning occurrences that have been added in this file are a bit misleading. 

Me, as a user, that I have intentionally omitted the 1D barcode support, I would not want to receive a warning message, let alone a warning that contains the word "Failed". 

Since this is a change that does not relate to the CMake update, I would suggest to remove the changes of this file, send a new PR and I will merge the rest. Then, we could open an issue to talk about the best approach on the compiler warning (if needed).